### ISSUE TITLE
[core] Logs from Checksum and Signing are a bit confusing when deploying to maven central

### ISSUE THREAD

**Issue by ttddyy** [jreleaser/jreleaser#1159]
I get these logs for checksum and signing while publishing the artifacts to maven central.
```
[INFO] Calculating checksums
[INFO]   [checksum] No files configured for checksum. Skipping
[INFO] Signing files
[INFO]   [sign] No files configured for signing. Skipping
```
This was initially confusing why no checksum/signing is happening because publishing to maven central requires adding checksum and signing artifacts; so I thought these checksum and signing should happen.

When I made the next deployment phase properly configured, the Deployment phase log includes `asc`, `md5`, etc files, so I understood checksum and signing have been done maybe in this phase.

```
[INFO] Deploying all staged artifacts
[INFO]   [nexus2] Deploying to maven-central
[INFO]   [nexus2] Lookup staging profile for net.ttddyy.observation
[INFO]   [nexus2] Creating a staging repository for net.ttddyy.observation
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar.asc
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar.md5
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar.sha1
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar.sha256
[INFO]   [nexus2]  - datasource-micrometer-1.0.1-javadoc.jar.sha512
...
```

It would be nice if there is some notion in either documentation or log that these steps don't need to perform when publishing artifacts to maven central.

**Comment by aalmiray** [comment]
Related to https://github.com/jreleaser/jreleaser/issues/1136

Checksum and signing steps affect artifacts defined in the `distributions` and `files` blocks, not staged artifacts defined in the `deploy` block.

**Comment by aalmiray** [comment]
Additionally, per the workflow explained at https://jreleaser.org/guide/latest/concepts/workflow.html#_deploy the `deploy` step itself does not depend on any other which means invoking `mvn deploy:deploy` won't  trigger `checksum` nor `sign`. However, `release` depends on both `deploy` and `upload` with the latter depending on `deploy` and `sign`, which is why you saw those messages.

**Comment by aalmiray** [comment]
Released in v1.5.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.5.0
